modules/3d/src/pointcloud/coders_obj.cpp

```diff
@@ -0,0 +1,112 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "coders_obj.hpp"
+#include <fstream>
+#include <iostream>
+#include <opencv2/core/utils/logger.hpp>
+#include "utils.hpp"
+
+namespace cv { namespace pc {
+
+std::unordered_set<std::string> ObjDecoder::m_unsupportedKeys;
+
+void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<std::vector<int32_t>> &indices)
+{
+    points.clear();
+    normals.clear();
+    indices.clear();
+
+    std::ifstream file(m_filename, std::ios::binary);
+    if (!file)
+    {
+        CV_LOG_WARNING(NULL, "Impossible to open the file: " + m_filename);
```

Comment:
> `+ m_filename`

Don't concatenate strings here.
Use `<< m_filename` instead.

---

> CV_LOG_WARNING

It should be error message: CV_LOG_ERROR